perun-core/src/main/java/cz/metacentrum/perun/core/impl/ExtSourcePerun.java

```diff
@@ -139,6 +139,9 @@ public static PerunBlImpl setPerunBlImpl(PerunBlImpl perun) {
 			}
 		}
 
+		//Null login is not allowed there
+		if(login == null) throw new InternalErrorException("Login can't be null.");
```

Comment:
It would be handy to log related richUser as well. It might help with debug if this error ever occurs.